sycl/test-e2e/KernelCompiler/sycl_imf.cpp

```diff
@@ -0,0 +1,81 @@
+//==--- sycl_imf.cpp --- kernel_compiler extension imf tests ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: aspect-usm_device_allocations
+// REQUIRES: (opencl || level_zero)
+
+// UNSUPPORTED: accelerator
+// UNSUPPORTED-INTENDED: while accelerator is AoT only, this cannot run there.
+
+// RUN: %{build} -o %t.out
+// RUN: %{l0_leak_check} %{run} %t.out
+
+#include <sycl/detail/core.hpp>
+#include <sycl/kernel_bundle.hpp>
+#include <sycl/usm.hpp>
+
+auto constexpr IMFSources = R"===(
+#include <sycl/sycl.hpp>
+#include <cmath>
+#include <sycl/ext/intel/math.hpp>
+
+extern "C" SYCL_EXTERNAL 
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(sycl::ext::oneapi::experimental::single_task_kernel)
+void imf_kernel(float *ptr) {
+  // cl_intel_devicelib_imf
+  //ptr[0] = sycl::ext::intel::math::sqrt(ptr[0] * 2);
+
+  // cl_intel_devicelib_imf_bf16
+  //ptr[1] = sycl::ext::intel::math::float2bfloat16(ptr[1] * 0.5f);
+}
```

Comment:
all the code in this kernel is commented out. Is that intentional?

Reply:
Ha, that explains why the values come back unchanged :)
Thank you for spotting that!